core/app/finders/spree/products/find.rb

```diff
@@ -0,0 +1,85 @@
+module Spree
+  module Products
+    class Find
+      def initialize(scope, params)
+        @scope = scope
+
+        @ids      = String(params[:ids]).split(',')
+        @price    = params[:price].try(:to_f)
+        @taxons   = String(params[:taxons]).split(',')
+        @name     = params[:name]
+        @options  = params[:options].try(:to_unsafe_hash)
+      end
+
+      def call
+        products = by_ids(scope)
+        products = by_price(products)
+        products = by_taxons(products)
+        products = by_name(products)
+        products = by_options(products)
+
+        products
+      end
+
+      private
+
+      attr_reader :ids, :price, :taxons, :name, :options, :scope
+
+      def ids?
+        ids.present?
+      end
+
+      def price?
+        price.present?
+      end
+
+      def taxons?
+        taxons.present?
+      end
+
+      def name?
+        name.present?
+      end
+
+      def options?
+        options.present?
+      end
+
+      def arel_condition(field, condition)
+        Arel.sql("#{field} = #{condition}")
+      end
+
+      def by_ids(products)
+        return products unless ids?
+
+        products.where(id: ids)
+      end
+
+      def by_price(products)
+        return products unless price?
+
+        products.joins(master: :prices).where(arel_condition("#{Spree::Price.table_name}.amount", price))
```

Comment:
@VygovskySergey we want to support price ranges, as specified in the swagger docs - in the API we will submit price min and price max values, eg. `price=10,50`